scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -1821,7 +1821,19 @@ class Router(formatOps: FormatOps) {
             Seq(Split(Space.orNL(noNL), 0))
         }
 
-      // Pat
+      // Union/Intersection types
+      case FormatToken(T.Ident(value), _, _)
```

Comment:
optional: should `Type.And` and `Type.Or` be derived from
```
object Type {
  trait BinaryType {
    def lhs: Type
    def rhs: Type
    val op: Type.Name
  }
}
```
and match
```
case FormatToken(ident: T.Ident, _, _) if leftOwner match {
  case x: Type.Binary => x.op.tokens.head eq ident // unless there's a better check
  case _ => false
} =>
```

Reply:
was just a thought. i frequently have them 😉

Reply:
I really appreciate the comments and the reviews, thank you!